### PR TITLE
update readme to clarify the merge function

### PR DESCRIPTION
**Pull request by xerocross**
As it is, the readme file on this repo is unclear on exactly what the library does.  For one thing, an object does not have descriptors.  It has properties, and the properties have descriptors.  I tried to make it clearer.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,7 +5,7 @@
 [![Build Status][travis-image]][travis-url]
 [![Test Coverage][coveralls-image]][coveralls-url]
 
-Merge objects using descriptors.
+Merge source object properties onto a destination object, and preserve descriptors.  Descriptors are configuration data associated to each property such as whether the property is `writable`, `enumerable`, and/or `configurable`.  This data will be preserved in the destination object.
```
Comment (dougwilson) [inline README.md]: Please hard wrap the lint to approx 80 chars. And use a single space instead of double space.

File: README.md [hunk 2]
```diff
@@ -25,15 +25,9 @@ animal.name === 'jon'
 
 ## API
 
-### merge(destination, source)
+### merge(Object destination, Object source [, bool overwrite])
```
Comment (dougwilson) [inline README.md]: The style here is that this is actual JavaScript. Please update the header to at least be valid JavaScript still 👍

File: README.md [hunk 3]
```diff
@@ -25,15 +25,9 @@ animal.name === 'jon'
...
 
-Redefines `destination`'s descriptors with `source`'s. The return value is the
-`destination` object.
-
-### merge(destination, source, false)
-
-Defines `source`'s descriptors on `destination` if `destination` does not have
-a descriptor by the same name. The return value is the `destination` object.
+Merge properties from `source` object into `destination` object, preserving each property's descriptors.  If `overwrite == false` then existing properties on `destination` will not be overwritten.  The default value of `overwrite` is true.  Returns the destination object.
```
Comment (dougwilson) [inline README.md]: Please hard wrap the lint to approx 80 chars. And use a single space instead of double space.

**Review by dougwilson — CHANGES_REQUESTED** [review]
Thanks, just a couple formatting nits to address.

File: README.md (hunk 2 above)
Comment (xerocross) [reply README.md]: ok.  Do you prefer to maintain two distinct definitions with/without the third argument?

Comment (dougwilson) [reply README.md]: You could put a variable instead as the third argument. But as long as it's obvious that the third argument can just be omitted in the function call.

File: README.md (hunk 1 above)
Comment (dougwilson) [reply README.md]: So maybe it should explain which of the source properties are going to be copied into the destination as well.

Comment (xerocross) [reply README.md]: Good point.  That would seem to be the object's "own" properties.

**Comment by xerocross** [comment]
I made those changes.  I think it's ready for review again.  Let me know if you want more changes.  (I'm unfamiliar with the PR system here on github.)

**Comment by dougwilson** [comment]
Nice! All looks good. I'm not really saying to change this just mainly curious: with such a small readme, does a large portion of it really need to be written twice?

**Comment by xerocross** [comment]
Fair point.  Do you want the repetition cut out at the top?  I think it needs to stay in the API description.

**Comment by dougwilson** [comment]
So IDK what I want changed around that, which is why I just just trying to understand the change vs asking for a specific change.

File: README.md [hunk 4]
```diff
@@ -5,7 +5,11 @@
 [![Build Status][travis-image]][travis-url]
 [![Test Coverage][coveralls-image]][coveralls-url]
 
-Merge objects using descriptors.
+Merge source object "own" properties onto a destination object, including 
+non-enumerable "own" properties, and preserve each property's descriptors. 
+Descriptors are configuration data associated to each property such as whether 
+the property is `writable`, `enumerable`, and/or `configurable`.  This data
```
Comment (dougwilson) [inline README.md]: So getters, setters, and the value is also copied over, which isn't listed. This (at least to me?) reads almost like the value of the property is not actually copied over. WDYT?

Also there is still a double space on this line by accident.

Comment (xerocross) [reply README.md]: Your right.  I could add a little more to clarify that.  Give me a moment.

**Comment by xerocross** [comment]
how's that?

**Comment by xerocross** [comment]
dammit hold on.  I wrote another double space.

**Comment by xerocross** [comment]
Do you have any more change requests on this?

**Comment by sindresorhus** [comment]
I have tried to clarify the behavior in the current readme.

**Pull request closed without merging**